protocol/v2/ssv/runner/runner.go

```diff
@@ -224,15 +224,14 @@ func (b *BaseRunner) basePartialSigMsgProcessing(
 
 // didDecideCorrectly returns true if the expected consensus instance decided correctly
 func (b *BaseRunner) didDecideCorrectly(prevDecided bool, decidedMsg *specqbft.SignedMessage) (bool, error) {
-	decided := decidedMsg != nil
-	decidedRunningInstance := decided && b.State.RunningInstance != nil && decidedMsg.Message.Height == b.State.RunningInstance.GetHeight()
-
-	if !decided {
+	if decidedMsg == nil {
 		return false, nil
 	}
-	if !decidedRunningInstance {
+
+	if b.State.RunningInstance == nil || b.State.RunningInstance.GetHeight() != decidedMsg.Message.Height {
```

Comment:
If we see the error, it won't be clear what condition was triggered. I'd split this into two statements like in https://github.com/bloxapp/ssv-spec/pull/342 but change the error text for the first one to indicate that it's nil

Reply:
https://github.com/bloxapp/ssv-spec/pull/372#issuecomment-2024676353

Reply:
https://github.com/bloxapp/ssv-spec/pull/372#issuecomment-2024888004